### PR TITLE
Feature/repeat pages

### PR DESCRIPTION
**Pull request by jenbutongit**
# Description

Repeatable pages support! This is currently a developer only feature (so no support in designer for it right now.)

- Repeatable pages can be flagged with "repeatableField" property which the field storing '_n_'. 
- These values are stored as an array of objects inside section. The objects key the values by page path `sectionName: [{/page-1: {...}, /page-2: {...}}]` to keep track of which pages have been "completed". Each array item is 1 "iteration" of the repeatable section.
- Support for summary
  - summary-detail has been split out into summary-row
  - Added isArray filter to detect if the section...is an array (repeated) 
- [x] waiting on a patch from @superafroman also 
- change circleci to trigger engine build + test

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Use dynamic.json inside `runner/test/cases` if you would like to "play" a repeatable form. 

- [x] Unit tested

# Checklist:

- [x] My code follows the [javascript standard style](https://standardjs.com/)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and versioning
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have rebased onto master and there are no code conflicts

### CODE REVIEW

File: engine/src/page.js [hunk 1]
```diff
@@ -333,22 +330,24 @@ class Page {
...
         return h.view(this.viewName, viewModel)
       }
 
+
       let update = this.getPartialMergeState(stateResult.value)
       if (this.repeatField) {
-        const updateValue = { [this.path]: update[this.section.name] }
-        const sectionState = state[this.section.name]
+        let updateValue = {[this.path]: update[this.section.name]}
+        let sectionState = state[this.section.name]
```
Comment (superafroman) [inline engine/src/page.js:342]: Is this an auto format thing? Should be `const` from what I can see?

File: engine/src/page.js [hunk 2]
```diff
@@ -333,22 +330,24 @@ class Page {
...
         if (!sectionState) {
           update = { [this.section.name]: [updateValue] }
-        } else if (!sectionState[num - 1]) {
+        } else if(!sectionState[num-1]) {
```
Comment (superafroman) [inline engine/src/page.js:345]: Auto formatting again? Spacing heere is good - don't we have a standard formatter?

Comment (jenbutongit) [reply engine/src/page.js:345]: maybe merge issue, sorry! Might be that engine doesn't have linter installed and/or not running on test. I'll add.

File: engine/src/page.js [hunk 3]
```diff
@@ -86,38 +92,65 @@ class Page {
     }).filter(v => !!v)
   }
 
-  getNextPage (state) {
-    let defaultLink = null
-    const nextLink = this.next.find(link => {
-      const { page, condition } = link
-      const value = page.section ? state[page.section.name] : state
+  getNextPage (state, suppressRepetition = false) {
```
[the remark this answers is not in the document]
Comment (jenbutongit) [reply engine/src/page.js:95]: JS is a lot nicer these days :)

**Comment by jenbutongit** [comment]
It looks like babel-eslint/eslint wont be supporting private methods until babel v8. https://github.com/babel/babel/issues/10752, https://github.com/babel/babel-eslint/pull/801. Ignoring for now.

**Pull request merged**